src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -478,23 +479,88 @@ def test_dag_names():
     assert builtin.dag_name == "builtin_sum_x_12"
 
 
+def test_node_digraph_no_graphviz(monkeypatch):
+    ''' Test that the function to get the graphviz Digraph class type returns
+    None if graphviz is not installed. We monkeypatch sys.modules to ensure
+    that it always appears that graphviz is not installed on this system. '''
+    monkeypatch.setitem(sys.modules, 'graphviz', None)
+    dag_class = _graphviz_digraph_class()
+    assert dag_class is None
+
+
 def test_node_dag_no_graphviz(tmpdir, monkeypatch):
-    '''test that dag generation does nothing if graphviz is not
-    installed. We monkeypatch sys.modules to ensure that it always
-    appears that graphviz is not installed on this system. '''
+    ''' Test that the dag generation returns None (and that no file is created)
+    when graphviz is not installed. We make this test independent of whether or
+    not graphviz is installed by monkeypatching sys.modules. '''
     monkeypatch.setitem(sys.modules, 'graphviz', None)
     _, invoke_info = parse(
         os.path.join(BASE_PATH, "1_single_invoke.f90"),
         api="dynamo0.3")
     psy = PSyFactory("dynamo0.3",
                      distributed_memory=False).create(invoke_info)
     invoke = psy.invokes.invoke_list[0]
-    schedule = invoke.schedule
     my_file = tmpdir.join('test')
-    schedule.dag(file_name=my_file.strpath)
+    dag = invoke.schedule.dag(file_name=my_file.strpath)
+    assert dag is None
     assert not os.path.exists(my_file.strpath)
 
 
+def test_node_dag_returns_digraph(tmpdir, monkeypatch):
```

Comment:
There are a few pylint errors here, some of which are fixable ...

C:513, 4: Class name "fake_digraph" doesn't conform to PascalCase naming style (invalid-name)
W:516,27: Redefining built-in 'format' (redefined-builtin)
W:516,27: Unused argument 'format' (unused-argument)
C:519, 8: Missing method docstring (missing-docstring)
W:519,23: Unused argument 'name' (unused-argument)
C:522, 8: Missing method docstring (missing-docstring)
W:522,37: Unused argument 'color' (unused-argument)
W:522,30: Unused argument 'name2' (unused-argument)
W:522,23: Unused argument 'name1' (unused-argument)
C:525, 8: Missing method docstring (missing-docstring)
W:525,25: Unused argument 'filename' (unused-argument)
W:508,34: Unused argument 'tmpdir' (unused-argument)
C:546, 4: Class name "fake_digraph" doesn't conform to PascalCase naming style (invalid-name)
W:549,27: Redefining built-in 'format' (redefined-builtin)
W:549,27: Unused argument 'format' (unused-argument)
E:601, 4: Unable to import 'graphviz' (import-error)

Reply:
Argh. Apologies. Should all be fixed now.